applications/plugins/Geomagic/CMakeLists.txt

```diff
@@ -1,36 +1,49 @@
-cmake_minimum_required(VERSION 2.8.12)
-project(Geomagic)
+cmake_minimum_required(VERSION 3.1)
+project(Geomagic VERSION 0.1)
 
 find_package(SofaHaptics REQUIRED)
-find_package(OpenHaptics)
+find_package(OpenHaptics REQUIRED)
+find_package(SofaOpenglVisual REQUIRED)
+
 if( NOT OPENHAPTICS_FOUND )
     message( SEND_ERROR "OpenHaptics library not found, GeomagicDriver cannot compile")
 endif()
 
-find_package(SofaOpenglVisual REQUIRED)
+set(GEOMAGIC_SRC_DIR src/Geomagic)
 
 set(HEADER_FILES
-      src/GeomagicDriver.h
-      src/initPlugin.h
-      src/config.h
+    ${GEOMAGIC_SRC_DIR}/config.h.in
+    ${GEOMAGIC_SRC_DIR}/GeomagicDriver.h
 )
 
-set(SOURCE_FILES
-      src/GeomagicDriver.cpp
-      src/initPlugin.cpp
+set(SOURCE_FILES	  
+    ${GEOMAGIC_SRC_DIR}/GeomagicDriver.cpp
+    ${GEOMAGIC_SRC_DIR}/initPlugin.cpp
 )
 
-set(SCENES_FILES
-      scenes/Geomagic-Demo.scn
-)
+set(README_FILES README.txt)
 
 include_directories( ${OPENHAPTICS_INCLUDE_DIR})
```

Comment:
This could also be changed to `target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${OPENHAPTICS_INCLUDE_DIR}>")`